app/src/main/java/nz/org/cacophony/birdmonitor/views/ManageRecordingsFragment.java

```diff
@@ -118,9 +118,13 @@ public void displayOrHideGUIObjects() {
     }
 
     public void uploadRecordings() {
-
-
-        if (!Util.isNetworkConnected(getActivity().getApplicationContext())) {
+        if (RecordAndUpload.isRecording) {
+            Toast.makeText(this.getContext(), R.string.upload_currently_recording, Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (Util.isAirplaneModeOn(getContext())) {
+            Util.disableFlightMode(getContext());
+        } else if (!Util.isNetworkConnected(getActivity().getApplicationContext())) {
```

Comment:
Don't we want this as an `if` not an `else if`? 
If Airplane mode was on you still want to check that you have an network connection after Airplane mode turns off?

Reply:
It actually waits for an internet connection in  Util.uploadFilesUsingUploadButton
so this check is just to disable flight mode, but if it wasn't enabled to see if we have an active connection.